### ISSUE TITLE
`@emotion/styled/base` imports don't work with native ES modules

### ISSUE THREAD

**Issue by matt-allan** [emotion-js/emotion#2413]
**Current behavior:**

I'm trying to use [native ECMAScript modules in Node.JS](https://nodejs.org/api/esm.html#esm_modules_ecmascript_modules) in a project that uses `@emotion/styled` and `@emotion/babel-preset-css-prop`.

When `@emotion/babel-preset-css-prop`  runs it injects an import from `@emotion/styled/base` which looks like this:

```js
import _styled from "@emotion/styled/base";
```

Since this is a directory import, and directory imports only work with CommonJS, it fails with an `ERR_UNSUPPORTED_DIR_IMPORT'` error.

**To reproduce:**

You can reproduce the issue with [this Github repository](https://github.com/matt-allan/emotion-styled-base-es-modules).

The main points are:

1. [Enable native ES modules](https://github.com/matt-allan/emotion-styled-base-es-modules/blob/b380ed7d63cfc0a6ca567c1d206059df663a69a3/package.json#L2)
2. [Use `@emotion/styled`](https://github.com/matt-allan/emotion-styled-base-es-modules/blob/b380ed7d63cfc0a6ca567c1d206059df663a69a3/src/index.js#L3)
3. [Transpile with `@emotion/babel-preset-css-prop`](https://github.com/matt-allan/emotion-styled-base-es-modules/blob/b380ed7d63cfc0a6ca567c1d206059df663a69a3/.babelrc#L11) and make sure [`@babel/preset-env` is configured with `{modules: false}`](https://github.com/matt-allan/emotion-styled-base-es-modules/blob/b380ed7d63cfc0a6ca567c1d206059df663a69a3/.babelrc#L6)
4. Try to run the resulting file with node: `node lib/index.js`

**Expected behavior:**

The import works.

**Environment information:**

- `react` version: 17.0.2
- `@emotion/react` version: 11.4.0
- `@emotion/styled` version: 11.3.0
- `@emotion/babel-preset-css-prop` version: 11.2.0
- `@emotion/babel-plugin` version: 11.3.0

I _think_ this is probably the same issue as #2121 since [Next.JS also sets `{modules: false}` for `@babel/preset-env`](https://nextjs.org/docs/advanced-features/customizing-babel-config) and is presumably trying to run Emotion in Node for SSR.

**Comment by matt-allan** [comment]
I'm able to workaround the issue with this Babel config:

```js
// babel.config.cjs

// Get the actual path to the ESM module, i.e. `@emotion/styled/base/dist/emotion-styled-base.esm.js`.
const path = require('path');
const emotionPkg = require('@emotion/styled/base/package.json');
const styledBaseImport = [
  '@emotion/styled/base/' + path.normalize(emotionPkg.module),
  'default',
];

module.exports = {
  presets: [
    [
      '@emotion/babel-preset-css-prop',
      {
        autoLabel: 'always',
        labelFormat: '[local]',
        useBuiltIns: false,
        throwIfNamespace: true,
        importMap: {
          '@emotion/styled': {
            default: {
              canonicalImport: ['@emotion/styled', 'default'],
              styledBaseImport,
            },
          },
        },
      },
    ],
  ]
}
```

**Comment by tobua** [comment]
Ran into the same issue with `@emotion/babel-plugin` and ES Modules. The `importMap` workaround did the trick:

```js
{
  plugins: [
    [
      '@emotion',
      {
        importMap: {
          '@emotion/styled': {
            default: {
              canonicalImport: ['@emotion/styled', 'default'],
              styledBaseImport: [
                '@emotion/styled/base/dist/emotion-styled-base.esm.js',
                'default',
              ],
            },
          },
        },
      },
    ],
  ],
}
```

**Comment by Andarist** [comment]
Actually, this wasn't fixed by #2819 but I would consider this issue here to be a duplicate of #2730 and this going to close this one here.
